app/services/price_policy_updater.rb

```diff
@@ -39,10 +40,27 @@ def assign_attributes
 
   def save
     ActiveRecord::Base.transaction do
-      @price_policies.all?(&:save) || raise(ActiveRecord::Rollback)
+      (@price_policies.all?(&:save) && save_product) || raise(ActiveRecord::Rollback)
     end
   end
 
+  def save_product
+    keys_from_built_rate_starts = []
+
+    rate_starts_params = permitted_product_params[:rate_starts_attributes].clone
+    rate_starts_params.values.each_with_index do |dr, index|
+      if dr[:min_duration].blank?
+        if dr[:id].present?
+          rate_starts_params["#{index}"].merge! ({ _destroy: '1' })
+        else
+          keys_from_built_rate_starts << "#{index}"
+        end
+      end
+    end
+
+    @product.update({ rate_starts_attributes: rate_starts_params.except(*keys_from_built_rate_starts) })
```

Comment:
A note for later, if we have time: I think we should have a separate method that assembles the appropriate data from params.  Then we should be able to pass it over to the price_policy:
```
def price_policy_attributes
  price_group_attributes(price_policy.price_group).merge(product_attributes)
end

def product_attributes
 keys_from_built_rate_starts = []
  ...
  { rate_starts_attributes: rate_starts_params.except(*keys_from_built_rate_starts) }
end

price_policy.assign_attributes(price_policy_attributes)

```